libraries/AdaptiveExpressions/FunctionUtils.cs

```diff
@@ -585,6 +585,39 @@ public static EvaluateExpressionDelegate ApplyWithError(Func<IReadOnlyList<objec
                 return (value, error);
             };
 
+        /// <summary>
+        /// Generate an expression delegate that applies function after verifying all children.
+        /// </summary>
+        /// <param name="function">Function to apply.</param>
+        /// <param name="verify">Function to check each arg for validity.</param>
+        /// <returns>Delegate for evaluating an expression.</returns>
+        public static EvaluateExpressionDelegate ApplyWithOptionsAndError(Func<IReadOnlyList<object>, Options, (object, string)> function, VerifyExpression verify = null)
+            =>
+            (expression, state, options) =>
+            {
+                object value = null;
+                string error = null;
+                IReadOnlyList<object> args;
+                (args, error) = EvaluateChildren(expression, state, options, verify);
+                if (error == null)
+                {
+                    try
+                    {
+                        (value, error) = function(args, options);
+                    }
+#pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
Always add a justification for a #pragma disable. Leave the rule description and add your justification in parenthesis so anyone that comes after you knows why you did it).

Example:
#pragma warning disable CA1031 // Do not catch general exception types (capture any exception and return it in the error)

Reply:
justification added